cmd/contour-operator.go

```diff
@@ -60,29 +61,24 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	mgrOpts := ctrl.Options{
 		Scheme:             scheme,
 		MetricsBindAddress: metricsAddr,
 		LeaderElection:     enableLeaderElection,
 		LeaderElectionID:   "0d879e31.projectcontour.io",
-	})
-	if err != nil {
-		setupLog.Error(err, "unable to start contour-operator")
-		os.Exit(1)
 	}
 
-	if err = (&contourcontroller.Reconciler{
-		Config: contourcontroller.Config{
-			ContourImage: contourImage,
-			EnvoyImage:   envoyImage,
-		},
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("Contour"),
-		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
```

Comment:
Should we update `controller/contour/suite_test.go` similarly? (Follow-up?)

Reply:
@Miciah I tried, but I get an import cycle error:
```
$ make test
/Users/daneyonhansen/code/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
controller/contour/suite_test.go
go vet ./...
package github.com/projectcontour/contour-operator/controller/contour (test)
	imports github.com/projectcontour/contour-operator/controller/manager
	imports github.com/projectcontour/contour-operator/controller/contour: import cycle not allowed in test
make: *** [vet] Error 1
```

See [my branch](https://github.com/danehans/contour-operator/tree/suite_test_refactor_mgr_pkg) for details.

Reply:
I created https://github.com/projectcontour/contour-operator/issues/114 to track this issue as a follow-up.